pkg/idl/doc.go

```diff
@@ -14,49 +14,126 @@ See the License for the specific language governing permissions and
+// used on any type of list (struct, scalar, ...).
 //
-// Using that tag will generate the following OpenAPI extension:
+// Using this tag will generate the following OpenAPI extension:
 //  "x-kubernetes-list-type": "atomic"
 //
 // Map
 //
-// The tag looks like this:
+// Example:
 //  +listType=map
 //
-// These lists are like maps. Order is preserved upon merge. Using the
-// map tag on a non-struct will report an error during the generation
-// step.
+// These lists are like maps in that their elements have a non-index key
+// used to identify them. Order is preserved upon merge. Using the map
+// tag on a list with non-struct elements will result in an error during
+// the generation step.
 //
-// Using that tag will generate the following OpenAPI extension:
+// Using this tag will generate the following OpenAPI extension:
 //  "x-kubernetes-list-type": "map"
 //
 // Set
 //
-// The tag looks like this:
+// Example:
 //  +listType=set
 //
-// Sets are lists that can't have multiple times the same value. Each
-// value must be a scalar or an atomic list and can't be map, set or struct.
+// Sets are lists that must not have multiple times the same value. Each
+// value must be a scalar.
 //
-// Using that tag will generate the following OpenAPI extension:
+// Using this tag will generate the following OpenAPI extension:
 //  "x-kubernetes-list-type": "set"
 type ListType string
+
+// ListMapKey annotates map lists by specifying the key used as the index of the map.
+//
+// This tag MUST only be used on lists that have the listType=map
+// attribute, or the generation step will fail. Also, the value
+// specified for this attribute must be a field, of a scalar type, of
+// the child structure (no nesting is supported).
+//
+// An example of how this can be used is shown in the ListType (map) example.
+//
+// Example:
+//  +listMapKey=name
+//
+// Using this tag will generate the following OpenAPI extension:
+//  "x-kubernetes-list-map-key": "name"
+type ListMapKey string
+
+// MapType annotates a map to further describe its topology. It may
+// have only one value: "atomic". Atomic means that the entire map is
+// considered as a whole, rather than as distinct values.
+//
+// By default, a map will be considered as a set of distinct values that
+// can be updated individually. This default WILL NOT generate any
+// openapi extension, as this will also be interpreted as the default
+// behavior in the openapi definition.
+//
+// This tag MUST only be used on maps, or the generation step will fail.
+//
+// Atomic
+//
+// Example:
+//  +mapType=atomic
+//
+// Atomic maps will be entirely replaced when updated. This tag may be
+// used on any map.
+//
+// Using this tag will generate the following OpenAPI extension:
+//  "x-kubernetes-map-type": "atomic"
+type MapType string
+
+// OpenAPIGen needs to be described.
+type OpenAPIGen string
+
+// Optional needs to be described.
+type Optional string
+
+// PatchStrategy needs to be described.
```

Comment:
These will be deprecated I think

Reply:
Yeah I still want to describe them as such